### PR TITLE
CRM-17684 Find Memberships with no auto-renew

### PR DESCRIPTION
**Pull request by sqweets**
…tatus

---
- CRM-17684: Find Memberships with no auto-renew
  https://issues.civicrm.org/jira/browse/CRM-17684

### CODE REVIEW

**Comment by sqweets** [comment]
For this issue: https://issues.civicrm.org/jira/browse/CRM-17684

**Comment by totten** [comment]
jenkins, ok to test

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by yashodha** [comment]
Jenkins test this please

File: CRM/Member/BAO/Query.php [hunk 1]
```diff
@@ -274,11 +302,12 @@ public static function whereClauseSingle(&$values, &$query) {
...
-          $query->_qill[$grouping][] = ts("Membership is Auto-Renew");
-        }
-        else {
-          $query->_where[$grouping][] = " civicrm_membership.contribution_recur_id IS NULL";
-          $query->_qill[$grouping][] = ts("Membership is NOT Auto-Renew");
+          $query->_qill[$grouping][] = ts("Membership is Auto-Renew and Cancelled");
+        } 
+        elseif ($value == 5) {
+          $query->_where[$grouping][] = " civicrm_membership.contribution_recur_id IS NOT NULL";
+          $query->_where[$grouping][] = " ccr.end_date IS NOT NULL";
```
Comment (yashodha) [inline CRM/Member/BAO/Query.php]: Shouldn't it check end date is in past instead?

Comment (seamuslee001) [inline CRM/Member/BAO/Query.php]: I would thought you need both i would imagine

Comment (yashodha) [inline CRM/Member/BAO/Query.php]: Right :)

**Comment by yashodha** [comment]
check for past end date included in #8549

**Comment by JoeMurray** [comment]
Hi Ellen, as Release Manager this month, I'm trying to recruit people to help pare down the backlog of almost 100 PRs, some going back to last summer. I'm wondering if you would be able to help QA another PR if I got someone to QA this PR?

**Comment by sqweets** [comment]
Sure, I can help with that and see how it goes.

**Pull request merged**